File: 이진영/추가문제/24일차/영단어 암기는 괴로워.js
const main = () => {
    let input = require('fs').readFileSync('dev/stdin')
        .toString().trim().split("\n").map(i => i.trim());

    const [n, m]= input[0].split(" ").map(Number);
    const words = input.slice(1).filter(i => i.length >= m);

    let freq = new Map();
    words.forEach(word => {
        if (!freq.has(word)) {
            freq.set(word, 1);
        } else {
            freq.set(word, freq.get(word) + 1);
        }
    })

    console.log([... new Set(words)].sort((a, b) => {
        if (freq.get(a) !== freq.get(b)) return freq.get(b) - freq.get(a);
        else if (a.length !== b.length) return b.length - a.length;
        else {
            if (a > b) {
                return 1;
            } else {
                return -1;
            }
        }
    }).join("\n"));
}

main();

File: 현세미/백준/IT기업 및 대기업 계열사 코테보면서 비슷했던 문제들/영단어 암기는 괴로워.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())
words = [input().rstrip() for _ in range(N)]

words = [word for word in words if len(word) >= M]

freq = {}
for word in words:
    freq[word] = freq.get(word, 0) + 1

unique_words = list(freq.keys())

#리스트를 제자리에서 정렬해주는 함수
unique_words.sort(key=lambda x: (-freq[x], -len(x), x))

print('\n'.join(unique_words))
